static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt

```diff
@@ -51,16 +56,27 @@ class AutoDisposeDetector: Detector(), SourceCodeScanner {
   override fun visitMethod(context: JavaContext, node: UCallExpression, method: PsiMethod) {
     val evaluator = context.evaluator
 
-    if (isReactiveType(evaluator, method)) {
-      context.report(ISSUE, node, context.getLocation(node), "Subscription not managed by AutoDispose.")
+    if (isReactiveType(evaluator, method) && isInScope(evaluator, node.getContainingUClass())) {
+      context.report(ISSUE, node, context.getLocation(node), "Always apply an AutoDispose " +
+          "scope before subscribing within defined scoped elements.")
     }
   }
 
+  private fun isInScope(evaluator: JavaEvaluator, psiClass: PsiClass?): Boolean {
+    psiClass?.let {
+      return evaluator.inheritsFrom(psiClass, "androidx.lifecycle.LifecycleOwner", false) ||
+          evaluator.inheritsFrom(psiClass, "com.uber.autodispose.ScopeProvider", false) ||
```

Comment:
let's expand this to also cover `Activity` and `Fragment` (old and new). Specifically let's try to use the same default coverage as the error prone check, and stick this in something a bit more DRY'd up (`defaultScopes.any { evaluator.inheritsFrom(psiClass, it) }`)